examples/fluids/src/setupstg_shur14.c

```diff
@@ -0,0 +1,333 @@
+#endif
+
+#ifndef min
+#define min(a,b)            (((a) < (b)) ? (a) : (b))
+#endif
+
+#endif  /* NOMINMAX */
+
+/*
+ * @brief Perform Cholesky decomposition on array of symmetric 3x3 matrices
+ *
+ * This assumes the input matrices are in order [11,22,33,12,13,23]. This
+ * format is also used for the output.
+ *
+ * @param[in] nprofs Number of matrices in Rij
+ * @param[in] Rij Array of the symmetric matrices [6,nprofs]
+ * @param[out] Cij Array of the Cholesky Decomposition matrices, [6,nprofs]
+ */
+static inline void CalcCholeskyDecomp(int nprofs,
+                                      const CeedScalar Rij[6][nprofs], CeedScalar Cij[6][nprofs]) {
+
+  CeedPragmaSIMD
+  for(int i=0; i<nprofs; i++) {
+    Cij[0][i] = sqrt(Rij[0][i]);
+    Cij[3][i] = Rij[3][i] / Cij[0][i];
+    Cij[1][i] = sqrt(Rij[1][i] - pow(Cij[3][i], 2) );
+    Cij[4][i] = Rij[4][i] / Cij[0][i];
+    Cij[5][i] = (Rij[5][i] - Cij[3][i]*Cij[4][i]) / Cij[1][i];
+    Cij[2][i] = sqrt(Rij[2][i] - pow(Cij[4][i], 2) - pow(Cij[5][i], 2));
+  }
+}
+
+
+/*
+ * @brief Get the number of rows for the PHASTA file at path
+ *
+ * Assumes that the first line of the file has the number of rows and columns
+ * as the only two entries, separated by a single space
+ *
+ * @param[in] path Path to the file
+ * @param[out] nrows Number of rows
+ */
+static inline PetscErrorCode GetNRows(const char path[PETSC_MAX_PATH_LEN],
+                                      int *nrows) {
+
+  PetscErrorCode ierr;
+  int ndims;
+  FILE *fp;
+  const int char_array_len = 512;
+  char line[char_array_len];
+  MPI_Comm comm = PETSC_COMM_WORLD;
+  char **array;
+
+  PetscFunctionBeginUser;
+  ierr = PetscFOpen(comm, path, "r", &fp); CHKERRQ(ierr);
+
+  if ( fgets(line, char_array_len, fp) != NULL) {
+    ierr = PetscStrToArray(line, ' ', &ndims, &array); CHKERRQ(ierr);
+    if (ndims > 1) {
+      *nrows = atoi(array[0]);
+    } else {
+      printf("%s does not contain shape of contained data in the first line", path);
+      PetscFunctionReturn(-1);
+    }
+    ierr = PetscStrToArrayDestroy(ndims, array); CHKERRQ(ierr);
+  }
+
+  ierr = PetscFClose(comm, fp);
+  PetscFunctionReturn(0);
+}
+
+/*
+ * @brief Read the STGInflow file and load the contents into stg_ctx
+ *
+ * Assumes that the first line of the file has the number of rows and columns
+ * as the only two entries, separated by a single space.
+ * Assumes there are 14 columns in the file
+ *
+ * Function calculates the Cholesky decomposition from the Reynolds stress
+ * profile found in the file
+ *
+ * @param[in] path Path to the STGInflow.dat file
+ * @param[inout] stg_ctx STGShur14Context where the data will be loaded into
+ */
+static inline PetscErrorCode ReadSTGInflow(const char path[PETSC_MAX_PATH_LEN],
+    STGShur14Context stg_ctx) {
+
+  PetscErrorCode ierr;
+  int ndims, dims[2];
+  FILE *fp;
+  const int char_array_len = 512;
+  char line[char_array_len];
+  MPI_Comm comm = PETSC_COMM_WORLD;
```

Comment:
Best to pass the communicator in as a function argument rather than guess inside.